boudicca.base/enricher/src/main/kotlin/base/boudicca/enricher/service/osm/OsmEnricher.kt

```diff
@@ -0,0 +1,218 @@
+                SemanticKeys.LOCATION_CITY_PROPERTY,
+                address["city"]?.asText()
+            )
+        }
+        if (tags != null) {
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_CONTACT_EMAIL_PROPERTY,
+                tags["contact:email"]?.asText()
+            )
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_CONTACT_PHONE_PROPERTY,
+                tags["contact:phone"]?.asText()
+            )
+            try {
+                builder.updatePropertyIfEmpty(
+                    event,
+                    SemanticKeys.LOCATION_URL_PROPERTY,
+                    UrlUtils.parse(tags["contacts:website"]?.asText())
+                )
+            } catch (_: IllegalArgumentException) {
+                logger.info("Could not map ${tags["contacts:website"].asText()} to URI")
+            }
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_WIKIPEDIA_PROPERTY,
+                tags["wikipedia"]?.asText()
+            )
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_WIKIDATA_PROPERTY,
+                tags["wikidata"]?.asText()
+            )
+        }
+
+        return builder.build()
+    }
+
+    private fun getOrSearchOsmId(event: StructuredEvent): String? {
+        // TODO maybe think about taking all osmIds and not only the first :)
+        val osmIds = event.getProperty(SemanticKeys.LOCATION_OSM_ID_PROPERTY)
+        val osmId = if (osmIds.isNotEmpty()) {
+            osmIds.first().second
+        } else {
+            return searchOsmId(event)
+        }
+        return osmId
+    }
+
+    private fun searchOsmId(event: StructuredEvent): String? {
+        val locationNames = event
+            .getProperty(SemanticKeys.LOCATION_NAME_PROPERTY)
+            .map(Pair<Key, String>::second)
+        if (locationNames.isEmpty()) {
+            return null
+        }
+
+        val locationQuery = locationNames
+            .joinToString(separator = ",")
+            .encodeURL()
+
+        val query = """$NOMINATIM_BASE_URL/search?accept-language=de,en&format=jsonv2&limit=1&q=$locationQuery"""
+        val response = fetcher.fetchUrl(query)
+
+        val json = mapper.readTree(response) as ArrayNode
+
+        return if (json.isEmpty) null
+        else {
+            val elem = json.first()
+            val type = elem["osm_type"]?.asText()
+            val id = elem["osm_id"]?.asText()
+            if (!type.isNullOrEmpty() && !id.isNullOrEmpty()) {
+                type[0] + id
+            } else {
+                null
+            }
+        }
+    }
+
+    private fun buildAddress(tags: JsonNode): String {
+        val street = ((tags["road"]?.asText() ?: "") + " " + (tags["house_number"]?.asText() ?: "")).trim()
+        val postcode = tags["postcode"]?.asText()
```

Comment:
constants?

Reply:
i would also not use constants for things which are only used in one place and are not completely arbitrary like magic numbers